torch/csrc/distributed/rpc/python_functions.cpp

```diff
@@ -131,13 +131,17 @@ std::shared_ptr<FutureMessage> pyRpcBuiltin(
     AutogradMetadata autogradMetadata(
         autogradContext.context_id(), autogradContainer.newAutogradMessageId());
     RpcWithAutograd rpcWithAutograd(
+        agent.getWorkerInfo().id_,
         MessageType::MESSAGE_WITH_AUTOGRAD_REQ,
         autogradMetadata,
         std::move(scriptCall));
 
     // Record autograd information for 'send'.
     addSendRpcBackward(
-        autogradContext, autogradMetadata, rpcWithAutograd.tensors());
+        autogradContext,
+        autogradMetadata,
+        rpcWithAutograd.tensors(),
+        agent.getWorkerInfo().id_);
```

Comment:
this is not right, should be `dst.workerInfo.workerId_`

Reply:
Lets fix this? Did the current unit tests catch this? If not, lets add a unit test that catches this issue.

Reply:
The current tests don't catch this since all of the cpp tests only test directly when `addSendRpcBackward` is called, but in the actual code, we call it from other functions. To test here, we'd have to expose the worker_ids set in python, similar to how the `_send_functions` and `recv_functions` are exposed in `autograd/init.cpp`.